mantidproject/mantid
# Move common Indirect/Inelastic code into new MantidQtWidgetsSpectroscopy library

### Description of work
This PR moves the common widgets and classes used across the MantidQtInterfacesIndirect and MantidQtInterfacesInelastic libraries into a new library called MantidQtWidgetsSpectroscopy. This change means that the Indirect library no longer needs to depend on the Inelastic library. This was bad practise to have a "plugin" library depend on another "plugin" library, and so the new arrangement is better because MantidQtWidgetsSpectroscopy is not a plugin library. This plugin-to-plugin dependency would previously cause problems such as the problem highlighted in this issue https://github.com/mantidproject/mantid/issues/37587

Both the Indirect and Inelastic libraries now depend on MantidQtWidgetsSpectroscopy .

Fixes #37591

### To test:
1. Open Workbench
2. Check there are 4 interfaces in each of the `Inelastic` and `Indirect` categories
3. Make sure each of these interfaces open ok
4. Check there is no warnings in the console output
5. Test the "Restrict data by name" option in the Settings Dialog
6. Repeat this test but for the following conda package:
```
mamba install -c mantid/label/spectroscopy_lib_test mantidworkbench
```
*This does not require release notes* because **this is an internal change**


### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
